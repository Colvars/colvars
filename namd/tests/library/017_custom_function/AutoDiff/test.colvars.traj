# step         c                                                 
           0   (  1.47736880982107e+01 ,  5.47598531906441e-01 ) 
          10   (  1.64732268956053e+01 , -1.86635726975950e+00 ) 
          20   (  1.65450803775141e+01 , -2.58120974253006e+00 ) 
