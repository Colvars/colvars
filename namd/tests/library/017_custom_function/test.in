colvarsTrajFrequency 10

colvar {
  name c

  # This is a 2-vector function of a 4-vector
  customFunction x
  customFunction cos(x) + r1 + r3 

  distance {
    name x
    group1 { atomNumbers 4 }
    group2 { atomNumbers 99 }
  }
  distanceVec {
    name r
    group1 { atomNumbers 50 }
    group2 { atomNumbers 60 }
  }
}

harmonic {
  colvars c
  centers (20, -5)
  forceConstant 100
}
