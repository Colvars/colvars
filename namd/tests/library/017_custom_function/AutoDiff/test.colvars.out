colvars: ----------------------------------------------------------------------
colvars: Please cite Fiorin et al, Mol Phys 2013:
colvars:  http://dx.doi.org/10.1080/00268976.2013.813594
colvars: in any publication based on this calculation.
colvars: SMP parallelism is available.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration from file "test.in":
colvars: # indexFile = "" [default]
colvars: # smp = on [default]
colvars: # analysis = off [default]
colvars: # colvarsTrajFrequency = 10
colvars: # colvarsRestartFrequency = 100 [default]
colvars: # colvarsTrajAppend = off [default]
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = on [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "c"
colvars:   Initializing a new "distance" component.
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "group1".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       Enabling scalable calculation for group "group1".
colvars:       Atom group "group1" defined, 1 atoms initialized: total mass = 12.011, total charge = -0.1.
colvars:       Initializing atom group "group2".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       Enabling scalable calculation for group "group2".
colvars:       Atom group "group2" defined, 1 atoms initialized: total mass = 12.011, total charge = 0.07.
colvars:     # forceNoPBC = off [default]
colvars:     # oneSiteSystemForce = off [default]
colvars:     # oneSiteTotalForce = off [default]
colvars:   Initializing a new "distanceVec" component.
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "group1".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       Enabling scalable calculation for group "group1".
colvars:       Atom group "group1" defined, 1 atoms initialized: total mass = 12.011, total charge = 0.51.
colvars:       Initializing atom group "group2".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       Enabling scalable calculation for group "group2".
colvars:       Atom group "group2" defined, 1 atoms initialized: total mass = 12.011, total charge = 0.51.
colvars:     # forceNoPBC = off [default]
colvars:     # oneSiteSystemForce = off [default]
colvars:     # oneSiteTotalForce = off [default]
colvars:   All components initialized.
colvars:   This colvar uses a custom function.
colvars:   Expecting colvar value of type n-dimensional vector of size 2.
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 1 [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = off [default]
colvars:   # subtractAppliedForce = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 1 in total.
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "harmonic" instance.
colvars:   # name = "harmonic1" [default]
colvars:   # colvars = { c }
colvars:   # outputEnergy = off [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # centers = { ( 20 , -5 ) }
colvars:   # targetCenters = { ( 20 , -5 ) } [default]
colvars:   # forceConstant = 100
colvars:   # targetForceConstant = 0 [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 1 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: Updating NAMD interface:
colvars: updating atomic data (0 atoms).
colvars: updating group data (4 scalable groups, 4 atoms in total).
colvars: Updating NAMD interface:
colvars: updating atomic data (0 atoms).
colvars: updating group data (4 scalable groups, 4 atoms in total).
colvars: Re-initialized atom group c:0/0. 1 atoms: total mass = 12.011.
colvars: Re-initialized atom group c:0/1. 1 atoms: total mass = 12.011.
colvars: Re-initialized atom group c:1/0. 1 atoms: total mass = 12.011.
colvars: Re-initialized atom group c:1/1. 1 atoms: total mass = 12.011.
colvars: The restart output state file will be "test.restart.colvars.state".
colvars: The final output state file will be "test.colvars.state".
colvars: Writing to colvar trajectory file "test.colvars.traj".
colvars: Updating NAMD interface:
colvars: updating atomic data (0 atoms).
colvars: updating group data (4 scalable groups, 4 atoms in total).
colvars: Re-initialized atom group c:0/0. 1 atoms: total mass = 12.011.
colvars: Re-initialized atom group c:0/1. 1 atoms: total mass = 12.011.
colvars: Re-initialized atom group c:1/0. 1 atoms: total mass = 12.011.
colvars: Re-initialized atom group c:1/1. 1 atoms: total mass = 12.011.
colvars: The restart output state file will be "test.restart.colvars.state".
colvars: The final output state file will be "test.colvars.state".
colvars: Saving collective variables state to "test.colvars.state".
